### PR TITLE
Guard OBJ at operators

### PR DESCRIPTION
**Pull request by zzmp**
There is no bounds checking in QVector (see https://github.com/qtproject/qtbase/blob/dev/src/corelib/tools/qvector.h#L424). Instead, this adds a anonymously namespaced function to throw a std::out_of_range exception in the case that something tries to access a nonexistent QVector element in the OBJReader, where a crash had been observed.

### CODE REVIEW

**Comment by zzmp** [comment]
See: https://app.asana.com/0/74051501495175/114042076387237.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 15d25e3b42a9a4a9aec63a1eba945d1240febdb3 available at: https://gist.github.com/35c9868154c4ff8f968f127b4cea0e08<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7712.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7712.dmg

File: libraries/fbx/src/OBJReader.cpp [hunk 1]
```diff
@@ -33,6 +33,17 @@ QHash<QString, float> COMMENT_SCALE_HINTS = {{"This file uses centimeters as uni
 
 const QString SMART_DEFAULT_MATERIAL_NAME = "High Fidelity smart default material name";
 
+namespace {
```
Comment (Atlante45) [inline libraries/fbx/src/OBJReader.cpp]: Why the anonymous namespace?

Comment (zzmp) [inline libraries/fbx/src/OBJReader.cpp]: It's not declared in any header files, and I don't want any unintended side-effects from linkage.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 80d04987a9ca5afc49bb64f62ea393377a2e0664 available at: https://gist.github.com/35c9868154c4ff8f968f127b4cea0e08<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7712.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7712.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: d28b12ec3cd67d566d02e393ac7fdfff11b0fe3c available at: https://gist.github.com/35c9868154c4ff8f968f127b4cea0e08<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7712.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7712.dmg

**Comment by OmegaHeron** [comment]
Windows 7 - 4 attempts to launch loading "standard" home scene - 4 crashes.  Bugsplat sent for all.  QA Fail for me.

File: libraries/fbx/src/OBJReader.cpp [hunk 2]
```diff
@@ -33,6 +33,16 @@ QHash<QString, float> COMMENT_SCALE_HINTS = {{"This file uses centimeters as uni
...
+template<class T>
+T& checked_at(QVector<T> vector, int i) {
```
Comment (zzmp) [inline libraries/fbx/src/OBJReader.cpp]: This QVector<T> should be by reference.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 62f1e8d1297847d073c81b4bc8815633d938c32d available at: https://gist.github.com/35c9868154c4ff8f968f127b4cea0e08<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7712.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7712.dmg

**Comment by OmegaHeron** [comment]
Tested Windows 7 - general smoke test
Standard HF home scene - good.
Playa - good (crash recorded, but, it was out of GPU RAM error and not this PR's problem)
Cellscience - good. (Navigation script buttons are still unreliable - but, again, not this PR's problem)
Built in entities test (place/move/edit/remove/persist) - good.
ATP Upload, place, edit - good.

Crashed on close after all that - bugsplat sent.

Can't repro crash on close - so will mark QA Pass

**Pull request merged**